test/functional/test_framework/test_wrapper.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+from test_framework.test_framework import BitcoinTestFramework
+import argparse
+from os.path import abspath, join
+from os import getpid
+
+"""Wrapper Class for BitcoinTestFramework.
+
+Provides the BitcoinTestFramework rpc & daemon process management
+functionality to external python projects.
+
+The TestWrapper class is a singleton class, which ensures that users only ever
+instantiate a single TestWrapper."""
+
+class TestWrapper:
+
+    class __TestWrapper(BitcoinTestFramework):
+
+        def set_test_params(self):
+            # This can be overriden in setup() parameter.
+            self.num_nodes=3
+
+        def run_test(self):
+            pass
+
+        def setup(self,
+            bitcoind=abspath(join(__file__ ,"../../../..") +  "/src/bitcoind"),
+            bitcoincli=None,
+            setup_clean_chain=True,
+            num_nodes=3,
+            network_thread=None,
+            rpc_timeout=60,
+            supports_cli=False,
+            bind_to_localhost_only=True,
+            nocleanup=False,
+            noshutdown=False,
+            cachedir=abspath(join(__file__ ,"../../../..") + "/test/cache"),
+            tmpdir=None,
+            loglevel='INFO',
+            trace_rpc=False,
+            port_seed=getpid(),
+            coveragedir=None,
+            configfile=abspath(join(__file__ ,"../../../..") + "/test/config.ini"),
+            pdbonfailure=False,
+            usecli = False,
+            perf = False,
+            randomseed = None):
```

Comment:
Have you tried changing the method signature to catch `*kwargs` and then set the defaults below, to avoid this huge function signature?

Reply:
Yes, it is much cleaner as you suggested. Thank you!